### DISCUSSION TITLE
how to send same message to multiple channels?

### DISCUSSION THREAD

**Discussion by macd2** [Rapptz/discord.py#6424] [Q&A]
i used the example of background task and simply added a loop to send same message to different channels:
```
import discord
import asyncio

class MyClient(discord.Client):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

        # create the background task and run it in the background
        self.bg_task = self.loop.create_task(self.my_background_task())

    async def on_ready(self):
        print('Logged in as')
        print(self.user.name)
        print(self.user.id)
        print('------')

    async def my_background_task(self):
        await self.wait_until_ready()
        channle_ids =  [1111,2222]
        while not self.is_closed():

           for channel_id in channle_ids:
                channel = self.get_channel(channel_id)              
                await channel.send(file=file, embed=embed)

            await asyncio.sleep(60) # task runs every 60 seconds


client = MyClient()
client.run('token')
```

I simply want to send the same file to different channels but it always only send to **the first id** in in my channle_ids list. 
what can i do?

**Comment by Rapptz** [comment]
That code is mostly correct but you're ignoring errors.

1. `channel` could resolve to `None`, this case isn't handled.
2. The `send` call could fail with `discord.HTTPException` this case isn't handled either.

**Reply by macd2** [reply]
yes absolutely i will handle the error. But the issue is even when i do use only:

```
channel1 = self.get_channel(id1)
await channel1.send(file=file, embed=embed)
channel2 = self.get_channel(id2)
await channel2.send(file=file, embed=embed)
```
it will only send the file to channle1 NOT channel 2 if i invers the order than only channel2 nit channel1

meaning for some  reason the bot is only processing the first send command and not a second or third even if i use 2 different channel instances in this case. Really frustrating any solution for this?
